Marlin/Marlin_main.cpp

```diff
@@ -9483,25 +9483,39 @@ inline void gcode_M999() {
 }
 
 #if ENABLED(SWITCHING_EXTRUDER)
-  inline void move_extruder_servo(uint8_t e) {
-    const int angles[2] = SWITCHING_EXTRUDER_SERVO_ANGLES;
-    MOVE_SERVO(SWITCHING_EXTRUDER_SERVO_NR, angles[e]);
-    safe_delay(500);
+  inline void move_extruder_servo(const uint8_t e) {
+    stepper.synchronize();
+    constexpr int16_t angles[] = SWITCHING_EXTRUDER_SERVO_ANGLES;
+    static_assert(COUNT(angles) / 2 >= E_STEPPERS, "SWITCHING_EXTRUDER_SERVO_ANGLES needs 2 angles per servo.");
```

Comment:
@thinkyhead, this test:
`static_assert(COUNT(angles) / 2 >= E_STEPPERS, "SWITCHING_EXTRUDER_SERVO_ANGLES needs 2 angles per servo.");`
is not valid for `EXTRUDERS 3`
because in this case - 
`E_STEPPERS` = (3 + 1) / 2 = 2
and 
`#define SWITCHING_EXTRUDER_SERVO_ANGLES { 0, 90 } `//we need only two angles - for T0 and T1
So:
2 / 2 < 2 and:
```
exit status 1
static assertion failed: SWITCHING_EXTRUDER_SERVO_ANGLES needs 2 angles per servo.
```
the same for `EXTURUDERS 5`
`E_STEPPERS` = (5 + 1) / 2 = 3
`#define SWITCHING_EXTRUDER_SERVO_ANGLES { 0, 90, 0, 90 }` //4 angles - for T0/T1, T2/T3
So:
4/2 < 3
____
`static_assert(COUNT(angles) == 2 || COUNT(angles) == 4, "SWITCHING_EXTRUDER_SERVO_ANGLES needs 2 angles per servo.");` 
will work here